### PR TITLE
Port corruption test to use custom env

### PR DESCRIPTION
**Pull request by riversand963**
Allow corruption_test to run on custom env loaded via
`Env::LoadEnv()`.

Test plan:
```
make corruption_test
./corruption_test
```

Also run on in-house custom env.

### CODE REVIEW

File: db/corruption_test.cc [hunk 1]
```diff
@@ -173,10 +188,48 @@ class CorruptionTest : public testing::Test {
     ASSERT_GE(max_expected, correct);
   }
 
+  void CorruptFile(const std::string& fname, int offset, int bytes_to_corrupt) {
```
Comment (zhichao-cao) [inline db/corruption_test.cc:191]: This is copied from testutil.cc?

Comment (zhichao-cao) [reply db/corruption_test.cc:191]: Where does this used?

Comment (riversand963) [reply db/corruption_test.cc:191]: I guess this function was not there when I created this PR on my local repo last year...Let me remove it if possible.

Comment (riversand963) [reply db/corruption_test.cc:191]: Weird, it is not used. Will remove it.

**Review by zhichao-cao — APPROVED** [review]
LGTM, thanks!

**Review by mrambacher — COMMENTED** [review]
I would strongly recommend that you also add support for MEM_ENV:
if (test_env_uri) {
 ... 
} else if (getenv("MEM_ENV") != nullptr) {
  env_guard_.reset(new MockEnv(base_env));
  base_env = env_guard.get();
}

With support for MEM_ENV, you will likely find that there are places that are attempting an action against the database but using the wrong env.

One of these days I am going to write a helper to do all of those settings in one swoop...

**Comment by riversand963** [comment]
> I would strongly recommend that you also add support for MEM_ENV:
>
SG. The purpose of this PR is to explain how to use `Env::LoadEnv`, thus it may be better just to keep it simple. We can add `MEM_ENV` in another PR.

**Pull request closed without merging**